OSDOCS-4438: Deploy Microshift with mirror registry (#64852)
openshift/openshift-docs: modules/microshift-mirroring-prereqs.adoc

// Module included in the following assemblies:
//
// * microshift/running_applications/microshift-deploy-with-mirror-registry.adoc

:_content-type: CONCEPT
[id="microshift-mirroring-prereqs_{context}"]
= Mirroring prerequisites

> Follow the instructions in the "Configuring credentials that allow images to be mirrored" link provided in the Additional resources to create a `~/.pull-secret-mirror.json` file including the user credentials for accessing the mirror.

Reviewer: The document I am reading suggests that you edit .pull-secret.json not create a file called .pull-secret-mirror.json

Am I following the wrong document ?
https://docs.openshift.com/container-platform/4.13/installing/disconnected_install/installing-mirroring-installation-images.html#installation-adding-registry-pull-secret_installing-mirroring-installation-images

Author: Same as above.

Reviewer 2: Step 2 in the procedure is creating a new file, which is then edited by appending the mirror-specific info.

For example, the following section is added to the pull secret file for the `microshift-quay:8443` mirror registry using `microshift:microshift` as username and password.

.Example mirror registry section for pull secret file

[source,terminal]
----
"<microshift-quay:8443>": { <1>
    "auth": "<bWljcm9zaGlmdDptaWNyb3NoaWZ0>",
    "email": "<microshift-quay@example.com>"
},
----

> <1> Replace <microshift-quay:8443> with the host name and port of your mirror registry server.

Reviewer 2: Everything should be replaced in this file: auth and email values included.
How can we adjust this section to reflect this better?

// Module included in the following assemblies:
//
// * microshift/running_applications/microshift-deploy-with-mirror-registry.adoc

:_content-type: CONCEPT
[id="microshift-configuring-mirroring-prereqs_{context}"]
= Configuring mirroring prerequisites

> You must create a container image registry credentials file that allows the mirroring of images from your Internet-connected mirror host to your air-gapped mirror. Follow the instructions in the "Configuring credentials that allow images to be mirrored" link provided in the Additional resources to create a `~/.pull-secret-mirror.json` file that includes the user credentials for accessing the mirror.

Reviewer: Can you put a sentence here that says specifically it needs to be present on the mirror registry host ?

Author: How about "Follow the instructions in the "Configuring credentials that allow images to be mirrored" link provided in the Additional resources to create a `~/.pull-secret-mirror.json` file _on the mirror registry host_ that includes the user credentials for accessing the mirror."

> You must create a container image registry credentials file that allows the mirroring of images from your Internet-connected mirror host to your air-gapped mirror. Follow the instructions in the "Configuring credentials that allow images to be mirrored" link provided in the Additional resources to create a `~/.pull-secret-mirror.json` file on the mirror registry host that includes the user credentials for accessing the mirror.

Reviewer 3: Suggestion:
```suggestion
You must create a container image registry credentials file that allows the mirroring of images from your internet-connected mirror host to your air-gapped mirror. Follow the instructions in "Configuring credentials that allow images to be mirrored" in the following "Additional resources" section to create a `~/.pull-secret-mirror.json` file on the mirror registry host that includes the user credentials for accessing the mirror.
```

> [id="microshift-example-mirror-pull-secret-entry"]

Reviewer 3: ```suggestion
[id="microshift-example-mirror-pull-secret-entry_{context}"]
```

== Example mirror registry pull secret entry

> For example, the following section is added to the pull secret file for the `microshift-quay:8443` mirror registry using `microshift:microshift` as username and password.

Reviewer 3: ```suggestion
For example, the following section is added to the pull secret file for the `microshift-quay:8443` mirror registry using `microshift:microshift` as the username and password.
```

.Example mirror registry section for pull secret file
[source,terminal]
----
"<microshift_quay:8443>": { <1>
    "auth": "<microshift_auth>", <2>
    "email": "<microshift_quay@example.com>" <3>
},
----

> <1> Replace <microshift-quay:8443> with the host name and port of your mirror registry server.
> <2> Replace <microshift_auth> with the user password.
> <3> Replace <microshift_quay@example.com> with the user email.

Reviewer 3: ```suggestion
<1> Replace `<microshift-quay:8443>` with the host name and port of your mirror registry server.
<2> Replace `<microshift_auth>` with the user password.
<3> Replace `<microshift_quay@example.com>` with the user email.
```
Not sure exactly, but also might need to escape the email address with a `\` so that it doesn't become hyperlinked/clickable.
